aws/src/main/java/org/apache/iceberg/aws/LakeFormationAwsClientFactory.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.aws;
+
+import java.util.Map;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import software.amazon.awssdk.auth.credentials.AwsCredentials;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.AwsSessionCredentials;
+import software.amazon.awssdk.http.urlconnection.UrlConnectionHttpClient;
+import software.amazon.awssdk.regions.PartitionMetadata;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.glue.model.GetTableRequest;
+import software.amazon.awssdk.services.glue.model.GetTableResponse;
+import software.amazon.awssdk.services.kms.KmsClient;
+import software.amazon.awssdk.services.lakeformation.LakeFormationClient;
+import software.amazon.awssdk.services.lakeformation.model.GetTemporaryGlueTableCredentialsRequest;
+import software.amazon.awssdk.services.lakeformation.model.GetTemporaryGlueTableCredentialsResponse;
+import software.amazon.awssdk.services.lakeformation.model.PermissionType;
+import software.amazon.awssdk.services.s3.S3Client;
+
+/**
+ * This implementation of AwsClientFactory is used if
+ * {@link org.apache.iceberg.aws.AwsProperties#GLUE_LAKEFORMATION_ENABLED} is set to true.
+ * It uses the default credential chain to assume role. Third-party engines can further extend this class
+ * to any custom credential setup. It extends AssumeRoleAwsClientFactory to reuse the assuming-role approach
+ * for all clients except S3 and KMS. If a table is registered with LakeFormation, the S3/KMS client will use
+ * LakeFormation vended credentials, otherwise it uses AssumingRole credentials.
+ * For using LakeFormation credential vending for a third-party query engine, see:
+ * https://docs.aws.amazon.com/lake-formation/latest/dg/register-query-engine.html
+ */
+public class LakeFormationAwsClientFactory extends AssumeRoleAwsClientFactory {
+
+  public static final String LF_AUTHORIZED_CALLER = "LakeFormationAuthorizedCaller";
+
+  private String glueCatalogId;
+  private String dbName;
+  private String tableName;
+  private String glueAccountId;
```

Comment:
nit: move this below `glueCatalogId` because they are glue related variables

Reply:
fixed. thanks